src/System.Private.CoreLib/src/System/Reflection/CustomAttribute.cs

```diff
@@ -1916,28 +1846,26 @@ private static void VerifyPseudoCustomAttribute(RuntimeType pca)
         #endregion
 
         #region Internal Static
-        internal static Attribute[] GetCustomAttributes(RuntimeType type, RuntimeType caType, out int count)
+        internal static RuntimeType.ListBuilder<Attribute> GetCustomAttributes(RuntimeType type, RuntimeType caType)
```

Comment:
It may be better to return the ListBuilder via `out` argument. It should be a bit more performant, avoid unnecessary ListBuilder struct copies.

Reply:
Ran before and after - looking good:
## Before In/Out:
|                               Method |       Mean |     Error |    StdDev | Gen 0/1k Op | Gen 1/1k Op | Gen 2/1k Op | Allocated Memory/Op |
|------------------------------------- |-----------:|----------:|----------:|------------:|------------:|------------:|--------------------:|
|               'Class: Hit (inherit)' | 1,750.8 ns | 14.313 ns | 13.388 ns |      0.1221 |           - |           - |               384 B |
|              'Class: Miss (inherit)' |   396.8 ns |  3.043 ns |  2.846 ns |           - |           - |           - |                   - |
|            'Class: Hit (no inherit)' | 1,618.1 ns | 15.698 ns | 14.684 ns |      0.1221 |           - |           - |               384 B |
|           'Class: Miss (no inherit)' |   247.7 ns |  1.756 ns |  1.643 ns |           - |           - |           - |                   - |
|     'Method Override: Hit (inherit)' | 2,757.6 ns | 27.861 ns | 24.698 ns |      0.2060 |           - |           - |               656 B |
|    'Method Override: Miss (inherit)' | 1,645.3 ns |  8.619 ns |  8.062 ns |      0.1221 |           - |           - |               384 B |
|  'Method Override: Hit (no inherit)' | 1,713.5 ns | 42.165 ns | 59.110 ns |      0.1259 |           - |           - |               400 B |
| 'Method Override: Miss (no inherit)' | 1,613.1 ns |  5.791 ns |  5.416 ns |      0.1221 |           - |           - |               384 B |
|         'Method Base: Hit (inherit)' | 1,652.3 ns |  8.609 ns |  8.053 ns |      0.1221 |           - |           - |               384 B |
|        'Method Base: Miss (inherit)' |   279.5 ns |  1.226 ns |  1.087 ns |           - |           - |           - |                   - |
|      'Method Base: Hit (no inherit)' | 1,679.5 ns | 24.290 ns | 21.533 ns |      0.1221 |           - |           - |               384 B |
|     'Method Base: Miss (no inherit)' |   260.9 ns |  5.194 ns |  5.557 ns |           - |           - |           - |                   - |


|                               Method |       Mean |     Error |    StdDev | Gen 0/1k Op | Gen 1/1k Op | Gen 2/1k Op | Allocated Memory/Op |
|------------------------------------- |-----------:|----------:|----------:|------------:|------------:|------------:|--------------------:|
|               'Class: Hit (inherit)' | 1,757.9 ns | 10.445 ns |  9.259 ns |      0.1221 |           - |           - |               384 B |
|              'Class: Miss (inherit)' |   386.1 ns |  2.424 ns |  2.267 ns |           - |           - |           - |                   - |
|            'Class: Hit (no inherit)' | 1,585.4 ns | 15.031 ns | 13.325 ns |      0.1221 |           - |           - |               384 B |
|           'Class: Miss (no inherit)' |   233.5 ns |  1.760 ns |  1.646 ns |           - |           - |           - |                   - |
|     'Method Override: Hit (inherit)' | 2,663.0 ns | 17.049 ns | 15.947 ns |      0.2060 |           - |           - |               656 B |
|    'Method Override: Miss (inherit)' | 1,618.0 ns | 10.219 ns |  9.559 ns |      0.1221 |           - |           - |               384 B |
|  'Method Override: Hit (no inherit)' | 1,602.1 ns | 13.484 ns | 11.260 ns |      0.1259 |           - |           - |               400 B |
| 'Method Override: Miss (no inherit)' | 1,599.2 ns |  7.126 ns |  6.666 ns |      0.1221 |           - |           - |               384 B |
|         'Method Base: Hit (inherit)' | 1,621.4 ns |  8.812 ns |  8.243 ns |      0.1221 |           - |           - |               384 B |
|        'Method Base: Miss (inherit)' |   275.0 ns |  1.786 ns |  1.671 ns |           - |           - |           - |                   - |
|      'Method Base: Hit (no inherit)' | 1,615.2 ns | 16.760 ns | 13.995 ns |      0.1221 |           - |           - |               384 B |
|     'Method Base: Miss (no inherit)' |   258.2 ns |  2.453 ns |  2.175 ns |           - |           - |           - |                   - |

(I think the top result is a fluke from a not-yet-shut-down compiler) - pushed the change in latest.